x-pack/plugins/spaces/public/copy_saved_objects_to_space/lib/process_import_response.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  SavedObjectsImportAmbiguousConflictError,
+  SavedObjectsImportConflictError,
+  SavedObjectsImportFailure,
+  SavedObjectsImportMissingReferencesError,
+  SavedObjectsImportResponse,
+  SavedObjectsImportSuccess,
+  SavedObjectsImportUnknownError,
+  SavedObjectsImportUnsupportedTypeError,
+} from 'src/core/public';
+
+export interface FailedImport {
+  obj: Omit<SavedObjectsImportFailure, 'error'>;
+  error:
+    | SavedObjectsImportConflictError
+    | SavedObjectsImportAmbiguousConflictError
+    | SavedObjectsImportUnsupportedTypeError
+    | SavedObjectsImportMissingReferencesError
+    | SavedObjectsImportUnknownError;
+}
+
+export interface ProcessedImportResponse {
+  success: boolean;
+  failedImports: FailedImport[];
+  successfulImports: SavedObjectsImportSuccess[];
+}
```

Comment:
Yeah, `SavedObjectsImportFailure` and `SavedObjectsImportSuccess` both come from Core.
`SavedObjectsImportFailure` is older and commingles the error with the rest of the object fields. So basically the entire point of this function is to "un-commingle" that into `FailedImport` 😅.

I added `SavedObjectsImportSuccess` later, but I was able to use it on the client side without any additional processing.

I can add this:
```
export type SuccessfulImport = FailedImport;
```
next time I'm in the code, agree it would make things clearer.

Reply:
Thanks for the explanation!